samples/image-create-and-upload-sample/image_create_and_upload_script.sh

```diff
@@ -0,0 +1,61 @@
+#!/usr/bin/env bash
+source config.sh
+ 
+function standardize_object_name() {
+  log "Formatting Object name $1"
+  local object_original_url=$1
+  local object_original_name=${object_original_url##*/}
+  local object_extension=""
+  local object_temp_name=""
+  if echo $object_original_name | grep -q -i rhcos; then
+    echo $object_original_name | grep 'qcow2.gz' >/dev/null
+    [ $? -ne 0 ] && error "Unsupported file format"
+    object_extension="ova.gz"
+    object_temp_name=${object_original_name%.*.*}
+  elif echo $object_original_name | grep -q -i rhel; then
+    echo $object_original_name | grep 'qcow2' >/dev/null
+    [ $? -ne 0 ] && error "Unsupported file format"
+    object_extension="ova.gz"
+    object_temp_name=${object_original_name%.*}
+  fi
+  IMAGE_NAME=$(echo $object_temp_name | sed -e 's/\.//g' -e 's/ppc64le//g' -e 's/openstack//g' -e 's/kvm//g' -e 's/---*//g' -e 's/_/-/g'|grep -o -E '[a-z]*-[0-9]{2}')-$(date +%m%d%Y)
+  OBJECT_NAME=$IMAGE_NAME.${object_extension}
+}
+
+function create_and_upload_rhel_image() {
+  standardize_object_name $RHEL_URL
+  RHEL_OBJECT_NAME=$OBJECT_NAME
+  RHEL_IMAGE_NAME=$IMAGE_NAME	
+  echo "Convert rhel image"
+  pvsadm image qcow2ova  --image-name $RHEL_IMAGE_NAME --image-url  $RHEL_URL --image-dist rhel  --rhn-user $RHEL_USERNAME  --rhn-password $RHEL_PASSWORD
+  echo "Uploading rhel image"
+  pvsadm image upload  --bucket $BUCKET_NAME --file $RHEL_OBJECT_NAME --bucket-region $BUCKET_REGION --resource-group $RESOURCE_GROUP
+  echo "Importing rhel Image"
+  pvsadm image import --pvs-instance-name $PVS_INSTANCE_NAME --bucket $BUCKET_NAME --object $RHEL_OBJECT_NAME --pvs-image-name $RHEL_IMAGE_NAME --bucket-region $BUCKET_REGION 
+}
+
+
+function create_and_upload_rhcos_image() {
+  standardize_object_name $RHCOS_URL
+  RHCOS_OBJECT_NAME=$OBJECT_NAME
+  RHCOS_IMAGE_NAME=$IMAGE_NAME
+  echo "Convert rhcos image"
+  pvsadm image qcow2ova  --image-name $RHCOS_IMAGE_NAME --image-url  $RHCOS_URL --image-dist coreos
+  echo "Uploading rhcos image"
+  pvsadm image upload  --bucket $BUCKET_NAME --file $RHCOS_OBJECT_NAME --bucket-region $BUCKET_REGION --resource-group $RESOURCE_GROUP
+  echo "Importing rhcos Image"
+  pvsadm image import --pvs-instance-name $PVS_INSTANCE_NAME --bucket $BUCKET_NAME --object $RHCOS_OBJECT_NAME --pvs-image-name $RHCOS_IMAGE_NAME --bucket-region $BUCKET_REGION
+}
+
+
+dnf install -y  qemu-img cloud-utils-growpart
+echo 'Initializing pvsadm tool !'
+version=`curl --silent "https://api.github.com/repos/ppc64le-cloud/pvsadm/releases/latest"|grep '"tag_name":'|sed -E 's/.*"([^"]+)".*/\1/'`
+curl -sL https://github.com/ppc64le-cloud/pvsadm/releases/download/$version/pvsadm-linux-ppc64le -o /usr/local/bin/pvsadm
+chmod +x /usr/local/bin/pvsadm
+pvsadm version
+
+export IBMCLOUD_API_KEY=$API_KEY
+
+create_and_upload_rhel_image
+create_and_upload_rhcos_image
```

Comment:
This will enforce to convert both rhel and rhcos every time, lets add some logic to convert only which is mentioned in the config.sh or some other mean.

and also CentOS is missing in the flow.

Reply:
Added the centos and checked if url is present or not